enterprise/server/remote_execution/snaploader/snaploader.go

```diff
@@ -18,6 +20,13 @@ import (
 // Key represents a cache key pointing to a snapshot
 // manifest.
 type Key struct {
+	// InstanceName is the remote instance name associated with the snapshot.
```

Comment:
Note that the `GroupID` is not added to the `Key` struct in this PR, because in the remote snapshot case, the group ID would instead be implicit in the JWT on the outgoing gRPC context. So instead I opted to take in the `ctx` in the `*filecacheKey` funcs.